auth/src/main/java/com/firebase/ui/auth/viewmodel/idp/LinkingSocialProviderResponseHandler.java

```diff
@@ -84,9 +85,14 @@ public void onComplete(@NonNull Task<AuthResult> task) {
                     .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
                         @Override
                         public void onComplete(@NonNull Task<AuthResult> task) {
-                            // I'm not sure why we ignore failures here, but this mirrors previous
-                            // behavior.
-                            handleSuccess(response, task.getResult());
+                            if (task.isSuccessful()) {
+                                handleSuccess(response, task.getResult());
+                            } else if (task.getException() instanceof
+                                    FirebaseAuthUserCollisionException){
+                                handleMergeFailure(credential);
+                            }
+                            // I'm not sure why we ignore other failures here, but this mirrors
```

Comment:
This comment is no longer relevant?

Reply:
Well we still ignore other failures though. We only deal w/ a merge failure. I'll remove it.